android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/MeasureReportViewModel.kt

```diff
@@ -407,24 +421,26 @@ constructor(
     }
 
     measureReports
-      .takeIf { it.all { it.type == MeasureReport.MeasureReportType.INDIVIDUAL } }
+      .takeIf {
+        it.all { measureReport -> measureReport.type == MeasureReport.MeasureReportType.INDIVIDUAL }
+      }
       ?.groupBy { it.subject.reference }
-      ?.forEach {
-        val reference = Reference().apply { reference = it.key }
+      ?.forEach { entry ->
+        val reference = Reference().apply { reference = entry.key }
         val subject =
-          fhirEngine.get(reference.extractType()!!, reference.extractId()).let {
-            when (it) {
-              is Patient -> it.nameFirstRep.nameAsSingleString
-              is Group -> it.name
+          fhirEngine.get(reference.extractType()!!, reference.extractId()).let { resource ->
+            when (resource) {
+              is Patient -> resource.nameFirstRep.nameAsSingleString
+              is Group -> resource.name
               else ->
                 throw UnsupportedOperationException(
-                  "${it.resourceType} as individual subject not allowed"
+                  "${resource.resourceType} as individual subject not allowed"
                 )
             }
           }
 
-        val indicators =
-          it.value.flatMap { report ->
+        val theIndicators =
+          entry.value.flatMap { report ->
             val formatted = formatSupplementalData(report.contained, report.type)
             val title = indicators.find { it.url == report.measure }?.title ?: ""
```

Comment:
this is probably expensive, especially since it's in a nested loop.

how about we create a map from indicator.url to indicator.title? so we can do this in constant time.

so somewhere outside both these loops, 
```kotlin
indicatorUrlToTitleMap = Map...
```

then here

```suggestion
            val title = indicatorUrlToTitleMap.getOrDefault(report.measure, "")
```